### PR TITLE
Support proxyroles for sn-platform

### PR DESCRIPTION
**Pull request by tuteng**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/templates/proxy/proxy-cluster.yaml [hunk 1]
```diff
@@ -149,7 +149,7 @@ spec:
       forwardAuthorizationCredentials: "true"
       authenticationProviders: {{ include "pulsar.authenticationProviders" . }}
 {{- include "pulsar.authConfiguration" . | indent 6 }}
-      superUserRoles: "{{ .Values.auth.superUsers.broker }}"
+      superUserRoles: "{{ .Values.auth.superUsers.proxy }}"
```
Comment (urfreespace) [inline charts/sn-platform/templates/proxy/proxy-cluster.yaml:152]: when the `proxyRolesEnabled` is false, also use `.Values.auth.superUsers.proxy`?

Comment (tuteng) [reply charts/sn-platform/templates/proxy/proxy-cluster.yaml:152]: I think it should have been a mistake before

**Comment by Huanli-Meng** [comment]
@tuteng do we need to update the docs for this PR?

**Comment by tuteng** [comment]
> @tuteng do we need to update the docs for this PR?

Yes

**Review by urfreespace — DISMISSED** [review]

File: charts/sn-platform/templates/broker/broker-cluster.yaml [hunk 2]
```diff
@@ -269,7 +269,7 @@ spec:
 {{- include "pulsar.authConfiguration" . | indent 6 }}
       superUserRoles: "{{ .Values.auth.superUsers.broker }}"
       {{- if .Values.auth.superUsers.proxyRolesEnabled }}
-      proxyRoles: {{ .Values.auth.superUsers.proxy }}
+      proxyRoles: .Values.auth.superUsers.proxyRoles
```
Comment (zerbitx) [inline charts/sn-platform/templates/broker/broker-cluster.yaml:272]: Should this be wrapped in `{{ }}`?

Comment (tuteng) [reply charts/sn-platform/templates/broker/broker-cluster.yaml]: Thanks, fixed

**Comment by ericsyh** [comment]
@maxsxu @urfreespace @zerbitx @fantapsody @ciiiii @FushuWang Pls take a review again, thx.

**Review by zerbitx — APPROVED** [review]

**Review by urfreespace — APPROVED** [review]

**Review by maxsxu — APPROVED** [review]

**Pull request merged**